File: sliding window/209. Minimum Size Subarray Sum.java
class Solution {
    public int minSubArrayLen(int target, int[] nums) {
        int i = 0;
        int j = 0;
        int sum = 0;
        int min = Integer.MAX_VALUE;
        while (j < nums.length) {
            sum += nums[j];
            while (sum >= target) {
                sum -= nums[i];
                min = Math.min(min, j - i + 1);
                i++;
            }
            j++;
        }
        return min == Integer.MAX_VALUE ? 0 : min;

    }
}

File: LeetCode/Sliding Window/209_Minimum Size Subarray Sum.py
class Solution:
    def minSubArrayLen(self, target: int, nums: List[int]) -> int:
        start=0
        end=0
        sum_target=0
        length=0
        minLength=len(nums)+1
        while end<len(nums):
            sum_target+=nums[end]
            while sum_target>=target:
                minLength=min(end-start+1,minLength)
                sum_target-=nums[start]
                start+=1
            end+=1
        if minLength==len(nums)+1:
            return 0

        return minLength

        